include/configuration_enum.h

```diff
@@ -0,0 +1,35 @@
+#ifndef CONFIGURATION_ENUM_H
+#define CONFIGURATION_ENUM_H
+
+#include <cstdint>
+
+namespace Configuration
+{
+
+enum device_t: uint8_t
+{
+  SERIAL,
+  RC,
+  AIRSPEED,
+  GNSS,
+  SONAR,
+  BATTERY_MONITOR,
+  BAROMETER,
+  MAGNETOMETER,
+  DEVICE_COUNT // make sure this is last
+};
+
+inline device_t& operator++(device_t &dev)
+{
+  uint8_t return_value = dev;
+  return_value++;
+  if (return_value > DEVICE_COUNT)
+    return_value--;
+  dev = static_cast<device_t>(return_value);
+  return dev;
+}
+}
+typedef uint8_t hardware_config_t;
+typedef Configuration::device_t device_t;
```

Comment:
Isn't the `Configuration::` unnecessary since we're already in the `Configuration` namespace?

Reply:
The purpose of this line is to allow using the type `device_t` without always having to type `Configuration::device_t`. My goal with making the namespace, but then adding this typedef was to keep the specific values (`Serial`, `RC`, etc.) out of the global namespace, but keep device_t in the global namespace.